rusoto/core/src/request.rs

```diff
@@ -170,21 +198,43 @@ impl fmt::Display for TlsError {
     }
 }
 
+/// TODO: docs
+pub struct DefaultTlsClient(Reactor);
+
+impl DispatchSignedRequest for DefaultTlsClient {
+    type Future = DefaultTlsClientFuture;
+
+    fn dispatch(&self, signed_request: SignedRequest) -> Self::Future {
+        DefaultTlsClientFuture(self.0.dispatch(signed_request))
+    }
+}
+
+/// TODO: docs
+pub struct DefaultTlsClientFuture(ReactorFuture);
+
+impl Future for DefaultTlsClientFuture {
+    type Item = HttpResponse;
+    type Error = HttpDispatchError;
+
+    fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
+        self.0.poll()
+    }
+}
+
 /// Helper method for creating an http client with tls.
 /// Makes a `hyper` client with `NativeTls` for HTTPS support.
-pub fn default_tls_client() -> Result<Client, TlsError> {
-    let ssl = match NativeTlsClient::new() {
-        Ok(ssl) => ssl,
-        Err(tls_error) => {
-            return Err(TlsError {
-                message: format!("Couldn't create NativeTlsClient: {}", tls_error),
-            })
-        }
-    };
-    let connector = HttpsConnector::new(ssl);
-    let mut client = Client::with_connector(Pool::with_connector(Default::default(), connector));
-    client.set_redirect_policy(RedirectPolicy::FollowNone);
-    Ok(client)
+pub fn default_tls_client() -> Result<DefaultTlsClient, TlsError> {
+    Reactor::new(|handle| {
+        let connector = match HttpsConnector::new(4, &handle) {
```

Comment:
You know we hardcode this 4 in the credentials crate too, which I did, but I wonder. Should it always be 4? Maybe you want to use more or less.

Reply:
Hmm, so for the credentials crate I'm not sure we'd need any DNS resolver threads at all. From what I can see, both uses of `make_request` target static IPs, in which case resolving it will just parse the IP and return (https://doc.rust-lang.org/src/std/net/addr.rs.html#811).

I did some work a couple of weeks ago to [allow custom executors for DNS resolution](https://github.com/hyperium/hyper/pull/1333) in hyper which we might be able to use in `rusoto_credential` to avoid any additional threads.

Coming back to this use of hyper, I hard-coded it to 4 in order to keep `default_tls_client` as simple as possible. Hyper itself also defaults to 4, and I think it's nice when you don't have to think about it upfront as a consumer of the library. When there is a requirement to tune the connector, I think it'd be sensible to just ask the consumers to construct a `Client` themselves with the desired config.

Keen to hear your thoughts!